src/entt/entity/handle.hpp

```diff
@@ -27,17 +27,39 @@ struct basic_handle {
         basic_registry<entity_type>
     >;
 
+    /**
+    * @brief Constructs a default invalid handle.
+    */
+    basic_handle() ENTT_NOEXCEPT
+        : basic_handle{null}
+    {}
+
+    /**
+    * @brief Constructs an invalid handle.
+    */
+    basic_handle(null_t) ENTT_NOEXCEPT
+        : basic_handle{nullptr, null}
+    {}
+
+    /**
+    * @brief Constructs a handle from another handle.
+    * @param other Another handle
+    */
+    basic_handle(const basic_handle &other) ENTT_NOEXCEPT
+        : basic_handle{other.reg, other.entt}
+    {}
+
     /**
      * @brief Constructs a handle from a given registry and entity.
      * @param ref An instance of the registry class.
      * @param value An entity identifier.
      */
-    basic_handle(registry_type &ref, entity_type value = null) ENTT_NOEXCEPT
-        : reg{&ref}, entt{value}
+    basic_handle(registry_type *registry, entity_type value = null) ENTT_NOEXCEPT
```

Comment:
Why a pointer instead of a reference? The goal here is exactly that of having a non-null pointer, that is, a reference.
In case the user has not a registry at hand, another constructor can be used.

Reply:
Good point. When you pass a registry you most likely want to pass a valid object.